crates/server/src/config/mod.rs

```diff
@@ -250,9 +249,17 @@ impl ZoneConfig {
     ///
     /// this is ony used on first load, if dynamic update is enabled for the zone, then the journal
     /// file is the actual source of truth for the zone.
-    pub fn get_file(&self) -> PathBuf {
-        // TODO: Option on PathBuf
-        PathBuf::from(self.file.as_ref().expect("file was none"))
+    pub fn get_file(&self) -> Option<PathBuf> {
```

Comment:
AFAICT this is currently not being used in the codebase so I don't know how this utility getter is meant to be used in practice but I think it may make sense to preserve what the underlying `StoreConfig` variant was and return an enum alongside the path, e.g.
``` rust
pub struct ZoneFile {
    pub path: PathBuf,
    pub kind: ZoneFileKind,
}

pub enum ZoneFileKind {
    PlainText,
    Sqlite,
}
```

Reply:
I also noticed this isn't used anywhere. Maybe one of the maintainers can shed some light on its purpose.

As I mentioned 
> to external users it previously generated a panic in valid None cases

so I wonder if anyone uses it at all

Reply:
Separately, I wonder if we should stop using the term `file` for plain_text, and instead refer to it as either `TxtZone` or just `Zone`, as [Wikipedia](https://en.wikipedia.org/wiki/Zone_file) refers to it? Most things are going to be files in this context, so it seems wrong to use a generic name for plain text. 

Since it was created by BIND alternatively it could be `BINDZoneFile`. Or `RFC1035Zone` or something?

Reply:
Please avoid further proliferation of UPPER acronyms -- the idiomatic spelling would be `Bind` or `Rfc`.

(I do agree with the sentiment but feel both the references to bind and the RFC are probably too verbose to this? Maybe just `ZoneFile`? Unadorned `Zone` seems a little too ambiguous for my taste.)

Reply:
I'm ok with just `Zone` (and agree on the proper naming style)